working/1426-extension-types/feature-specification.md

```diff
@@ -0,0 +1,948 @@
+invariants associated with the extension type. We could write forwarding
+members in the extension body, but using show/hide can have the same
+effect, and it is much more concise and convenient.*
+
+We use the phrase _extension show/hide part_, or just _show/hide part_ when
+no doubt can arise, to denote a phrase derived from
+`<extensionShowHidePart>`. Similarly, an `<extensionShowClause>` is known
+as an _extension show clause_, and an `<extensionHideClause>` is known as
+an _extension hide clause_, similarly abbreviated to _show clause_ and
+_hide clause_.
+
+The show/hide part specifies which instance members of the on-type are
+available for invocation on a receiver whose type is the given extension
+type.
+
+If the show/hide part is empty, no instance members except the ones
+declared for `Object?` can be invoked on a receiver whose static type is
+the given extension type.
+
+If the show/hide part is a show clause listing some identifiers and types,
+invocation of an instance member is allowed if its basename is one of the
+given identifiers, or it is the name of a member of the interface of one of
+the types. Instance members declared for `Object?` can also be invoked.
+
+If the show/hide part is a hide clause listing some identifiers and types,
+invocation of an instance member is allowed if it is in the interface of
+the on-type and _not_ among the given identifiers, nor in the interface of
+the specified types.
+
+If the show/hide part is a show clause followed by a hide clause, then the
+available instance members is computed by first computing the set of
+included instance members specified by the show clause as described above,
+and then removing instance members from that set according to the hide
+clause, as described above.
+
+In a show or hide clause, it is possible that an
+`<extensionShowHideElement>` is an identifier that is the basename of a
+member of the interface of the on-type, and it is also the name of a type
+in scope. In this case, the name shall refer to the member.
+
+*A conflict is unlikely because type names in general are capitalized, and
+member names start with a lower-case letter. Some type names start with a
+lower-case letter, too (e.g., `int` and `dynamic`), but those names do not
+occur frequently as member names. Should a conflict arise anyway, a
+work-around is to import the shadowed type `T` with a prefix `p` and put
+`p.T` in the show or hide clause.*
+
+A compile-time error occurs if a hide or show clause contains an identifier
+which is not the basename of an instance member of the on-type. A
+compile-time error occurs if a hide or show clause contains a type which is
+not among the types that are implemented by the on-type of the extension.
+
+A compile-time error occurs if a member included by the show/hide part has
+a basename which is also the basename of a member declaration in the
+extension type.
+
+*For instance, if an extension `E` contains a declaration of a method named
```

Comment:
I think you mean to say "an extension `E` with a hide clause contains", right?  Otherwise you must always hide `toString` if you wish to override it, which seems off to me.

Reply:
Right, done.